example/main.go

```diff
@@ -43,42 +43,41 @@ func main() {
 	resolveTOTPSecretReference(client, item.VaultID, item.ID, "TOTP_onetimepassword")
 	sharelink := generateItemSharing(client, item.VaultID, item.ID)
 	fmt.Println(sharelink)
+	archiveItem(client, item.VaultID, item.ID)
 	deleteItem(client, item.VaultID, item.ID)
 }
 
 func listVaultsAndItems(client *onepassword.Client, vaultID string) {
 	// [developer-docs.sdk.go.list-vaults]-start
-	vaults, err := client.Vaults().ListAll(context.Background())
+	vaults, err := client.Vaults().List(context.Background())
 	if err != nil {
 		panic(err)
 	}
-	for {
-		vault, err := vaults.Next()
-		if errors.Is(err, onepassword.ErrorIteratorDone) {
-			break
-		} else if err != nil {
-			panic(err)
-		}
-
+	for _, vault := range vaults {
 		fmt.Printf("%s %s\n", vault.ID, vault.Title)
 	}
 	// [developer-docs.sdk.go.list-vaults]-end
 
 	// [developer-docs.sdk.go.list-items]-start
-	items, err := client.Items().ListAll(context.Background(), vaultID)
+	overviews, err := client.Items().List(context.Background(), vaultID)
 	if err != nil {
 		panic(err)
 	}
-	for {
-		item, err := items.Next()
-		if errors.Is(err, onepassword.ErrorIteratorDone) {
-			break
-		} else if err != nil {
-			panic(err)
-		}
-		fmt.Printf("%s %s\n", item.ID, item.Title)
+	for _, overview := range overviews {
+		fmt.Printf("%s %s\n", overview.ID, overview.Title)
 	}
 	// [developer-docs.sdk.go.list-items]-end
+
+	// [developer-docs.sdk.go.list-archived-items]-start
```

Comment:
Can we call this snippet something like "using item list filters"? I think this is also what we should aim to name it in the docs, stating out that the filtering available today is by states.